multy_core/src/golos/golos_account.cpp

```diff
@@ -0,0 +1,233 @@
+/* Copyright 2018 by Multy.io
+ * Licensed under Multy.io license.
+ *
+ * See LICENSE for details
+ */
+
+#include "multy_core/src/golos/golos_account.h"
+
+#include "multy_core/common.h"
+#include "multy_core/src/api/key_impl.h"
+#include "multy_core/src/hash.h"
+#include "multy_core/src/utility.h"
+#include "multy_core/src/exception.h"
+#include "multy_core/src/exception_stream.h"
+#include "multy_core/src/ec_key_utils.h"
+
+#include "wally_core.h"
+#include "wally_crypto.h"
+
+#include <cassert>
+
+namespace
+{
+const size_t GOLOS_KEY_HASH_SIZE = 4;
+const char GOLOS_PUBLIC_KEY_STRING_PREFIX[] = "GLS";
+const uint8_t GOLOS_KEY_PREFIX[] = {0x80};
+
+const BlockchainType GOLOS_MAIN_NET{BLOCKCHAIN_GOLOS, BLOCKCHAIN_NET_TYPE_MAINNET};
+} // namespace
+
+namespace multy_core
+{
+namespace internal
+{
+
+class GolosPublicKey : public PublicKey
+{
+public:
+    typedef std::vector<uint8_t> KeyData;
+    GolosPublicKey(KeyData data)
+        : m_data(std::move(data))
+    {}
+
+    ~GolosPublicKey()
+    {
+    }
+
+    PublicKeyPtr clone() const override
+    {
+        return make_clone(*this);
```

Comment:
That would not work as expected since that would create a copy of PublicKey-data only. Some extra trickery is required, like CRTP:
```
template <T>
class PublicKeyBase : public PublicKey
{
    // ...
    PublicKeyPtr clone() const override
    {
         return make_clone(*reinterpret_cast<T*>(this));
    }
};
```
Or similar.